### ISSUE TITLE
Monitor/Window ContentScale below 1.0?

### ISSUE THREAD

**Issue by christianparpart** [glfw/glfw#1578]
My content scale on Linux / KDE desktop is set to 1.0, however, using GLFW3, my application gets reported 0.72, and that makes my font sizes on my apps look smaller than other apps.

Why is GLFW reporting smaller content scales, IoW: what could I do to counter-measure this, if it's not a bug in GLFW but on my end (however, I couldn't find anything in the docs helping me out here).

**Comment by elmindreda** [comment]
What is the output of `xprop -root | grep -o 'Xft.dpi[^n]*'` on your machine?

**Comment by christianparpart** [comment]
> What is the output of `xprop -root | grep -o 'Xft.dpi[^n]*'` on your machine?

Hey @elmindreda , thx for your reply. Sadly the above command won't yield anything, as thre is no `Xft.dpi` substring in the output.

So I'll give you the whole part:

```
_NET_WM_USER_TIME(CARDINAL) = 3029632
_NET_WM_ICON_NAME(UTF8_STRING) = 
_KDE_NET_WM_DESKTOP_FILE(UTF8_STRING) = "org.kde.kwin"
PULSE_COOKIE(STRING) = "a96a9e5de2566f905029702dc3686b8024de0adbb10579f545b25fa0668947a1465f63ae66112d5c0a4ee37a0ae54a3f012d92efbe137c89a897b459121d4156c3cf6f5b446b3302fd11024b085d4eda6b85cd5acc8fc01827667cf3fe723e3db3bd8182ecc502e825a2c265edc27ab75172899b26daa590e064441738697bf540000a504577c343dc343b3979244facae0b205c98bb2f463a41e110443e61ed54c8bec0be33cc3c96c060c3b7dbccf3b0e401dd765290a0aee7b30032a8c17d7663dee7b4b1270664e6182bad4f928c744a27faf65d649d74c6de77983da7bf84ec9017be8588acefd6cf70255dfe4d9134743147f7ab79e3073d8c57e5425a"
PULSE_SERVER(STRING) = "{1c415912bd074c3a83b18627e4000bcb}unix:/run/user/1000/pulse/native"
PULSE_ID(STRING) = "1000@1c415912bd074c3a83b18627e4000bcb/1791"
_NET_WORKAREA(CARDINAL) = 0, 0, 1920, 1024, 0, 0, 1920, 1024, 0, 0, 1920, 1024, 0, 0, 1920, 1024
_KDE_SLIDE(_KDE_SLIDE) = 0x0
_KDE_PRESENT_WINDOWS_GROUP(_KDE_PRESENT_WINDOWS_GROUP) = 0x0
_KDE_PRESENT_WINDOWS_DESKTOP(_KDE_PRESENT_WINDOWS_DESKTOP) = 0x0
_KDE_KWIN_KSCREEN_SUPPORT(CARDINAL) = 0
_KDE_WINDOW_HIGHLIGHT(_KDE_WINDOW_HIGHLIGHT) = 0x0
_KDE_NET_WM_BACKGROUND_CONTRAST_REGION(_KDE_NET_WM_BACKGROUND_CONTRAST_REGION) = 0x0
_KDE_NET_WM_BLUR_BEHIND_REGION(_KDE_NET_WM_BLUR_BEHIND_REGION) = 0x0
GDK_VISUALS(INTEGER) = 43, 130
AT_SPI_BUS(STRING) = "unix:abstract=/tmp/dbus-9eDzuItjmT,guid=fd4c136d7951beea10fa46bf5d9f05a5"
_NET_CLIENT_LIST_STACKING(WINDOW): window id # 0x240000b, 0x480000a, 0x4a00030, 0x2e00006, 0x2e00086, 0x3e00001, 0x4a00007, 0x2400019
_NET_CLIENT_LIST(WINDOW): window id # 0x240000b, 0x2400019, 0x3e00001, 0x480000a, 0x4a00007, 0x4a00030, 0x2e00006, 0x2e00086
_NET_DESKTOP_GEOMETRY(CARDINAL) = 1920, 1080
_NET_ACTIVE_WINDOW(WINDOW): window id # 0x4a00007
_NET_CURRENT_DESKTOP(CARDINAL) = 0
_NET_DESKTOP_VIEWPORT(CARDINAL) = 0, 0, 0, 0, 0, 0, 0, 0
_NET_NUMBER_OF_DESKTOPS(CARDINAL) = 4
_NET_DESKTOP_LAYOUT(CARDINAL) = 0, 2, 2, 0
_NET_DESKTOP_NAMES(UTF8_STRING) = "Desktop 1", "Desktop 2", "Desktop 3", "Desktop 4"
_NET_SUPPORTING_WM_CHECK(WINDOW): window id # 0x1e00007
_NET_SUPPORTED(ATOM) = _NET_SUPPORTED, _NET_SUPPORTING_WM_CHECK, _NET_CLIENT_LIST, _NET_CLIENT_LIST_STACKING, _NET_NUMBER_OF_DESKTOPS, _NET_DESKTOP_GEOMETRY, _NET_CURRENT_DESKTOP, _NET_DESKTOP_NAMES, _NET_ACTIVE_WINDOW, _NET_WORKAREA, _NET_DESKTOP_LAYOUT, _NET_CLOSE_WINDOW, _NET_RESTACK_WINDOW, _NET_SHOWING_DESKTOP, _NET_WM_MOVERESIZE, _NET_MOVERESIZE_WINDOW, _NET_WM_NAME, _NET_WM_VISIBLE_NAME, _NET_WM_DESKTOP, _NET_WM_WINDOW_TYPE, _NET_WM_WINDOW_TYPE_NORMAL, _NET_WM_WINDOW_TYPE_DESKTOP, _NET_WM_WINDOW_TYPE_DOCK, _NET_WM_WINDOW_TYPE_TOOLBAR, _NET_WM_WINDOW_TYPE_MENU, _NET_WM_WINDOW_TYPE_DIALOG, _NET_WM_WINDOW_TYPE_UTILITY, _NET_WM_WINDOW_TYPE_SPLASH, _KDE_NET_WM_WINDOW_TYPE_OVERRIDE, _NET_WM_STATE, _NET_WM_STATE_MODAL, _NET_WM_STATE_MAXIMIZED_VERT, _NET_WM_STATE_MAXIMIZED_HORZ, _NET_WM_STATE_SHADED, _NET_WM_STATE_SKIP_TASKBAR, _NET_WM_STATE_SKIP_PAGER, _KDE_NET_WM_STATE_SKIP_SWITCHER, _NET_WM_STATE_HIDDEN, _NET_WM_STATE_FULLSCREEN, _NET_WM_STATE_ABOVE, _NET_WM_STATE_BELOW, _NET_WM_STATE_DEMANDS_ATTENTION, _NET_WM_STATE_STAYS_ON_TOP, _NET_WM_STATE_FOCUSED, _NET_WM_STRUT, _NET_WM_STRUT_PARTIAL, _NET_WM_ICON_GEOMETRY, _NET_WM_ICON, _NET_WM_PID, _NET_WM_PING, _NET_WM_USER_TIME, _NET_STARTUP_ID, _NET_WM_FULLSCREEN_MONITORS, _NET_WM_ALLOWED_ACTIONS, _NET_WM_ACTION_MOVE, _NET_WM_ACTION_RESIZE, _NET_WM_ACTION_MINIMIZE, _NET_WM_ACTION_SHADE, _NET_WM_ACTION_MAXIMIZE_VERT, _NET_WM_ACTION_MAXIMIZE_HORZ, _NET_WM_ACTION_FULLSCREEN, _NET_WM_ACTION_CHANGE_DESKTOP, _NET_WM_ACTION_CLOSE, _NET_FRAME_EXTENTS, _KDE_NET_WM_FRAME_STRUT, _KDE_NET_WM_TEMPORARY_RULES, _NET_WM_FULL_PLACEMENT, _KDE_NET_WM_ACTIVITIES, _KDE_NET_WM_SHADOW, _NET_WM_OPAQUE_REGION
KWIN_RUNNING(KWIN_RUNNING) = 0x1
_ORG_KDE_KWIN_DBUS_SERVICE(UTF8_STRING) = "org.kde.KWin"
_QT_SETTINGS_TIMESTAMP_:0(_QT_SETTINGS_TIMESTAMP_:0) = 0x0, 0x0, 0x0, 0x0, 0x0, 0x25, 0x84, 0x8f, 0x2, 0xa4, 0xd3, 0x7c, 0x0
KDE_SESSION_VERSION(CARDINAL) = 5
KDE_FULL_SESSION(STRING) = "true"
_XKB_RULES_NAMES(STRING) = "evdev", "pc105", "us", "", ""
XFree86_has_VT(INTEGER) = 1
XFree86_VT(INTEGER) = 1
Xorg_Seat(STRING) = "seat0"
```

While this is the output when setting the scale to 1.0, the following I'll get when scale is set to 1.1:

```
RESOURCE_MANAGER(STRING) = "Xft.dpi:\t106\n"
```

Whereas the next one is for scale = `1.5`:

```
RESOURCE_MANAGER(STRING) = "Xft.dpi:\t144\n"
```
The way to set the content scaling used was simply to go to KDE's system settings -> Hardware -> Display and Monitor -> Scale.

**Comment by elmindreda** [comment]
Thank you, that was very helpful!

This should be fixed now with 75294462b354d147aa5914c30e3005551cc04e4a. It seems KDE removes the `Xft.dpi` value when setting the scale to 1 in some circumstances.
